src/node.h

```diff
@@ -69,6 +69,8 @@ struct cmark_node {
 
   cmark_strbuf string_content;
 
+  cmark_syntax_extension *extension;
```

Comment:
Nope, an extension might (and some of mine do) create a regular code block for a different syntax for example, cmark needs to know who to ask if they can be closed.